src/operators/expression/array/map.ts

```diff
@@ -16,21 +16,16 @@ export function $map(
   expr: { input: RawArray; as: string; in: AnyVal },
   options?: Options
 ): AnyVal {
-  const inputExpr = computeValue(obj, expr.input, null, options) as RawArray;
-  assert(
-    isArray(inputExpr),
-    `$map 'input' expression must resolve to an array`
-  );
-
-  const asExpr = expr.as || "this";
-  const inExpr = expr.in;
+  const input = computeValue(obj, expr.input, null, options) as RawArray;
+  assert(isArray(input), `$map 'input' expression must resolve to an array`);
 
-  // HACK: add the "as" expression as a value on the object to take advantage of "resolve()"
-  // which will reduce to that value when invoked. The reference to the as expression will be prefixed with "$$".
-  // But since a "$" is stripped of before passing the name to "resolve()" we just need to prepend "$" to the key.
```

Comment:
I think this is worth preserving for context.

Reply:
Just saw this -- thanks for the merge. Should I create another PR to restore this, or are you okay with it being removed now? (I noticed it wasn't present in the same logic in filter)

Reply:
I won't object to another PR, but also happy to add it before the release :)